crates/store/src/db/sql/mod.rs

```diff
@@ -1135,6 +1135,55 @@ pub fn unconsumed_network_notes(
     Ok((notes, page))
 }
 
+/// ...
+pub fn unconsumed_network_notes_for_network_account(
+    transaction: &Transaction,
+    network_account_id_prefix: NetworkAccountPrefix,
+    block_num: BlockNumber,
+    mut page: Page,
+) -> Result<(Vec<NoteRecord>, Page)> {
+    assert_eq!(
+        NoteExecutionMode::Network as u8,
+        0,
+        "Hardcoded execution value must match query"
+    );
+
+    // Select the rowid column so that we can return a pagination token.
+    //
+    // rowid column _must_ come after the note fields so that we don't mess up the
+    // `NoteRecord::from_row` call.
+    let mut stmt = transaction.prepare_cached(&format!(
+        "
+        SELECT {}, rowid
+        FROM notes
+        LEFT JOIN note_scripts ON notes.script_root = note_scripts.script_root
+        WHERE
+            execution_mode = 0 AND consumed = FALSE AND
+            block_num = ? AND rowid >= ?
+        ORDER BY rowid
+        LIMIT ?
+        ",
```

Comment:
@Mirko-von-Leipzig Do we need to make a schema change in order to correlate `notes` table with network account id prefixes?

Really early draft, try ignore most of it.

Reply:
I don't think that's required. You should be able to convert the prefix into the note tag and filter for the account using that.

If you check the `TryFrom<NoteTag> for NetworkAccountPrefix` you'll notice that the note tag is a simple cast once the pre-conditions are met. So going in reverse is essentially a no-op at the moment.

So you can filter for the matching notes by adding `tag = {network_account_id_prefix}`.